Apsitvarkom.UnitTests/DataAccess/PollutedLocationsFileStoreTests.cs

```diff
@@ -0,0 +1,103 @@
+﻿using Apsitvarkom.DataAccess;
+using static Apsitvarkom.Models.Enumerations;
+
+namespace Apsitvarkom.UnitTests.DataAccess;
+
+public class PollutedLocationsFileStoreTests
+{
+    [Test]
+    [TestCase("9719d4ef-5cde-4370-a510-53af84bdede2", -181.12311, LocationSeverityLevel.High, "2015-05-16T05:50:06", 100)]
+    public void GetAllPollutedLocationsDTO_SomePropertiesMissing_DeserializingSuccessful_MissingPropertiesSetToNull(string id, double longitude, LocationSeverityLevel severity, string creationTime, int progress)
```

Comment:
This got me thinking. If I understand correctly, if a json object is missing some properties, the deserialization will still succeed, but some of the fields will be null. In the DTO, we say that these properties `= null!` which is not true in this case. I think it would be best to have DTO properties nullable. E.g. transform `public string Id { get; init; } = null!;` into `public string? Id { get; init; }`

Reply:
Unless there's a way to make the `JsonSerializer.Deserialize` throw an exception if it does not find **all** of the fields. But I am not sure if that's possible.

Reply:
When the DTO properties are nullable, check how automapper handles when mapping `PollutedLocationDTO` -> `PollutedLocation` and some of the fields are null. It would be best if the `Map` method throws in that case.

Reply:
> Unless there's a way to make the `JsonSerializer.Deserialize` throw an exception if it does not find **all** of the fields. But I am not sure if that's possible.

https://github.com/dotnet/runtime/issues/37483 - seems like there is no such way as of yet.

Reply:
Let's create a new task for that: adding explicit validation for DTO objects.